sentry-ruby/lib/sentry/net/http.rb

```diff
@@ -78,6 +78,12 @@ def extract_request_info(req)
 
         result
       end
+
+      def uri_parser
+        URI::RFC2396_PARSER
+      rescue NameError
+        URI::DEFAULT_PARSER
+      end
```

Comment:
What do you think about defining our own constant conditionally instead of adding another method to the callstack? ie we could have this:

```ruby
URI_PARSER = URI.const_defined?(RFC2396_PARSER) ? URI::RFC2396_PARSER : URI::DEFAULT_PARSER
```

Then just use it in the extract_request_info `URI_PARSER.escape(...)`